R/run_eggla_gwas.R

```diff
@@ -586,61 +619,48 @@ run_eggla_gwas <- function(
           idcol = "trait_model"
         ),
         old = function(x) sub("^#", "", x)
-      )[TEST %in% "ADD" & !is.na(P), -c("TEST")]
+      )[
+        TEST %in% "ADD" & !is.na(P),
+        -c("TEST", "T_STAT", "OBS_CT")
+      ]
 
       output_results_file <- sprintf("%s.results.gz", results_file)
 
       if (!quiet) message(sprintf("[%s] Writing annotated results file ...", basename(vcf)))
-      if (use_info) {
-        output_dt <- data.table::merge.data.table(
-          y = results[j = A1 := NULL],
-          x = annot,
-          by.y = c("ID", "CHROM", "POS", "ALT", "REF"),
-          by.x = c("SNPID", "CHR", "POS", "EFFECT_ALLELE", "NON_EFFECT_ALLELE"),
-          all.y = TRUE
-        )
-        data.table::setcolorder(
-          x = output_dt,
-          neworder = c(
-            "trait_model",
-            "SNPID", "STRAND", "BUILD",
-            "CHR", "POS", "EFFECT_ALLELE", "NON_EFFECT_ALLELE",
-            "N", "N0", "N1", "N2", "EAF",
-            "BETA", "SE", "P",
-            "HWE_P", "CALL_RATE",
-            "IMPUTED", "INFO_TYPE", "INFO",
-            "T_STAT", "OBS_CT", "ERRCODE"
-          )
-        )
-      } else {
-        output_dt <- data.table::setnames(
-          x = results[j = A1 := NULL][
-            j = `:=`(
-              STRAND = strand,
-              BUILD = build
-            )
-          ],
-          old = c("ID", "CHROM", "POS", "ALT", "REF"),
-          new = c("SNPID", "CHR", "POS", "EFFECT_ALLELE", "NON_EFFECT_ALLELE")
-        )
-        data.table::setcolorder(
-          x = output_dt,
-          neworder = intersect(
-            x = c(
-              "trait_model",
-              "SNPID", "STRAND", "BUILD",
-              "CHR", "POS", "EFFECT_ALLELE", "NON_EFFECT_ALLELE",
-              "N", "N0", "N1", "N2", "EAF",
-              "BETA", "SE", "P",
-              "HWE_P", "CALL_RATE",
-              "IMPUTED", "INFO_TYPE", "INFO",
-              "T_STAT", "OBS_CT", "ERRCODE"
+      data.table::fwrite(
+        x = data.table::setnames(
+          x = Reduce(
+            f = function(x, y) merge(x, y, by = intersect(colnames(x), colnames(y))),
+            x = list(
+              data.table::fread(sprintf("%s.vmiss", results_file))[
+                j = list(
+                  `#CHROM`,
+                  ID,
+                  F_MISS,
+                  N = OBS_CT - MISSING_CT
+                )
+              ],
+              data.table::fread(sprintf("%s.afreq", results_file))[j = -c("OBS_CT")],
+              data.table::fread(sprintf("%s.hardy", results_file))[
+                j = list(
+                  `#CHROM`,
+                  ID,
+                  REF = A1, # weirdly A1 for HWE is ref in PLINK2
+                  HWE_P = P,
+                  HOM_REF_N = HOM_A1_CT,
+                  HET_REF_N = HET_A1_CT,
+                  TWO_ALT_N = TWO_AX_CT
+                )
+              ],
```

Comment:
For Hardy-Weinberg results, A1 appears to be REF in my toy dataset, could you double check this behaviour? @annihei @mboissel @Ning-L

Reply:
I don't think hardy has "effect allele", so maybe A1 is no t informative with this test?

Reply:
I still need it to merge the results for Hardy-Weinberg with the GWAS results.
So, its does not matter what A1 means, but I need to ensure that A1 is always reference allele and AX is the alternative, otherwise there will be issues when adding HWE results to GWAS results.

Reply:
OK yes I didn't realise you don't get ALT and REF in the hardy results until I tried to replicate using your code. What if you just add those two in your output ?
``
--hardy cols=+alt,+ref 
``

Reply:
Does this allow change the A1/AX in the other columns computed by `--hardy`?
Because if the columns still contains A1/AX, I still have to rename to ensure this A1 is not read as A1 from GWAS by anyone looking at the results file.

Reply:
It does not change A1 and AX. I am trying to check what results this provides now as it does not match with those calculated from the hardcall genotypes GT nor the posterior probabilities GP.  Can we get the desired output just using dosage DS?

Reply:
For N why don't we use ``N = OBS_CT - MISSING_DOSAGE_CT`` instead of MISSING_CT? 
``--missing vcols=+pos,+ref,+alt,+nmissdosage``

Reply:
Because, I did not know^^
Although, does it make a difference when we are using the `"dosage=DS"` specifier? Especially, since it basically says to PLINK to only look at dosage, thus in this case `MISSING_CT` should be the same as `MISSING_DOSAGE_CT`.

Reply:
I've been confusing with this as well.. See the row 4 :
> vmiss %>% filter(MISSING_CT > 0)
 #CHROM      POS          ID REF ALT MISSING_DOSAGE_CT MISSING_CT OBS_CT
     1:     21  9414087  21:9414087   G   A                0          1   2903
     2:     21  9414249  21:9414249   T   C                 0          1   2903
     3:     21  9415411  21:9415411   A   T                 0          1   2903
     4:     21  9415943  21:9415943   G   A                 0      2871   2903
     5:     21  9418624  21:9418624   C   T                 0          1   2903

Reply:
Edit: there was a missing column in my output I think we use the F_MISS
> vmiss %>% filter(MISSING_CT > 0)
        #CHROM      POS          ID REF ALT MISSING_DOSAGE_CT MISSING_CT OBS_CT      F_MISS
     1:     21  9414087  21:9414087   G   A                 0          1   2903 0.000344471
     2:     21  9414249  21:9414249   T   C                 0          1   2903 0.000344471
     3:     21  9415411  21:9415411   A   T                 0          1   2903 0.000344471
     4:     21  9415943  21:9415943   G   A                 0       2871   2903 0.988977000
     5:     21  9418624  21:9418624   C   T                 0          1   2903 0.000344471

Reply:
@annihei As you can see from your table, `F_MISS = 1 - CALL_RATE` based on `MISSING_CT` and `OBS_CT`.

Reply:
I am resolving this thread since HWE was removed.